hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/AbstractRecordStore.java

```diff
@@ -73,6 +73,7 @@ protected AbstractRecordStore(MapContainer mapContainer, int partitionId) {
         this.mapServiceContext = mapContainer.getMapServiceContext();
         NodeEngine nodeEngine = mapServiceContext.getNodeEngine();
         this.serializationService = nodeEngine.getSerializationService();
+        this.metadataStore = new MetadataStore(mapContainer, serializationService, JsonMetadataInitializer.INSTANCE, partitionId);
```

Comment:
Seems `metadataStore` creation only required when `MetadataPolicy` is `CREATE_ON_UPDATE`. It is redundant to create it in other cases.